### ISSUE TITLE
Bad SafeDisc sectors are incorrectly "Correctly retried" on ASUS BW-16D1HT

### ISSUE THREAD

**Issue by TheRogueArchivist** [claunia/DiscImageChef#732]
### Version

5.3.0

### Tested debug version?

- [X] Yes

### Which operating systems have you used?

- [X] Windows
- [ ] Linux
- [ ] macOS
- [ ] Other

### What is the architectural bit size you're using?

- [ ] 32-bit
- [X] 64-bit
- [ ] Unsure or unknown

### What processor are you using?

- [X] An Intel or AMD
- [ ] An ARM or Apple Silicon
- [ ] Unsure or unknown

### Device manufacturer

ASUS

### Device model

BW-16D1HT (FW 3.02+3.10)

### Bus the device uses to attach to the computer

- [ ] Parallel ATA
- [ ] Serial ATA
- [ ] SCSI (any)
- [ ] ATAPI (mark above if parallel or serial)
- [X] USB
- [ ] FireWire
- [ ] PCMCIA
- [ ] SecureDigital
- [ ] MultiMediaCard

### What were you doing when it failed?

- [X] I was dumping media (disk, tape, etc)...
- [ ] I was retrieving media (disk, tape, etc) information...
- [ ] I was scanning media (disk, tape, etc)...
- [ ] I was retrieving device information...

### Description

When dumping on an ASUS BW-16D1HT, some sectors are incorrectly "successfully" retried. This is most easy to notice with SafeDisc protected discs, I'm not currently sure if/how it applies to other protections or legitimate read issues. The dump goes as expected at first, with the expected slowdown during the protected range, and with the errors being trimmed afterwards. Once the dump reaches the "retry sectors" stage, every single sector is correctly retried. Due to this, instead of the sectors being 00 filled, they appear to be full of some kind of garbage data. It appears to me to be some kind of drive caching FW issue, where certain attempts at rereading result in garbage data being returned. Especially suspect to me is that the individual sectors are not successfully read during the "trim" stage, only in the "retry" stage. I also verified that my PX-760A dumped the disc as expected, meaning that this is drive issue. Confirmed on FW versions 3.02 and 3.10.

### Exact command line used

aaru m dump D: Sweet3.aaruf

### Expected behavior

The disc to have unreadable sectors due to SafeDisc.

### Actual behavior

All the intentionally bad sectors are incorrectly "correctly retried".

### Output of command execution with debug output enabled

```shell
5.3.1
Copyright c 2011-2022 Natalia Portillo

Output image format: Aaru Format.
Device in database since 12/10/2021 22:25:50.

Reading full TOC
Building track map...
Checking if drive supports PQ subchannel reading...
Checking if drive supports full raw subchannel reading...
Full raw subchannel reading supported...
Drive returns subchannel in BCD...
Calculating pregaps, can take some time...
Reading ATIP
Reading Disc Information
Reading PMA
Reading Session Information
Reading CD-Text from Lead-In
Solving lead-outs...
Detecting disc type...
Checking mode for track 1...
Track 1 is MODE1
Reading 16 sectors at a time.
Device reports 19527 blocks (47802096 bytes).
Device can read 16 blocks at a time.
Device reports 2448 bytes per logical block.
SCSI device type: MultiMediaDevice.
Media identified as CDROM.
Setting flags for track 1...
Track 1 starts at LBA 0 and ends at LBA 19526
No audio tracks, disabling offset fix.
Drive reading offset is 24 bytes (6 samples).
Disc offsets is 0 bytes (0 samples)
Setting speed to MAX for data reading.
Setting index 1 for track 1 to LBA 0.
Reading sector 19520 of 19527 (0.718 MiB/sec.)
Dump finished in 54.3952982 seconds.
Average dump speed 1012.567 KiB/sec.
Average write speed 5853.724 KiB/sec.
Trimming skipped sectors
Trimming sector 10687
Trimming finished in 690.8611407 seconds.
Correctly retried sector 888 in pass 1.
Correctly retried sector 895 in pass 1.
Correctly retried sector 910 in pass 1.
Correctly retried sector 919 in pass 1.
Correctly retried sector 922 in pass 1.
Correctly retried sector 929 in pass 1.
Correctly retried sector 968 in pass 1.
Correctly retried sector 977 in pass 1.
Correctly retried sector 995 in pass 1.
Correctly retried sector 1037 in pass 1.
Correctly retried sector 1050 in pass 1.
Correctly retried sector 1057 in pass 1.
Correctly retried sector 1060 in pass 1.
Correctly retried sector 1096 in pass 1.
Correctly retried sector 1103 in pass 1.
Correctly retried sector 1108 in pass 1.
Correctly retried sector 1123 in pass 1.
Correctly retried sector 1126 in pass 1.
Correctly retried sector 1181 in pass 1.
Correctly retried sector 1182 in pass 1.
Correctly retried sector 1192 in pass 1.
Correctly retried sector 1201 in pass 1.
Correctly retried sector 1240 in pass 1.
Correctly retried sector 1255 in pass 1.
Correctly retried sector 1258 in pass 1.
Correctly retried sector 1267 in pass 1.
Correctly retried sector 1286 in pass 1.
Correctly retried sector 1298 in pass 1.
Correctly retried sector 1337 in pass 1.
Correctly retried sector 1344 in pass 1.
Correctly retried sector 1391 in pass 1.
Correctly retried sector 1403 in pass 1.
Correctly retried sector 1417 in pass 1.
Correctly retried sector 1418 in pass 1.
Correctly retried sector 1430 in pass 1.
Correctly retried sector 1469 in pass 1.
Correctly retried sector 1476 in pass 1.
Correctly retried sector 1483 in pass 1.
Correctly retried sector 1488 in pass 1.
Correctly retried sector 1535 in pass 1.
Correctly retried sector 1545 in pass 1.
Correctly retried sector 1557 in pass 1.
Correctly retried sector 1558 in pass 1.
Correctly retried sector 1568 in pass 1.
Correctly retried sector 1611 in pass 1.
Correctly retried sector 1616 in pass 1.
Correctly retried sector 1623 in pass 1.
Correctly retried sector 1634 in pass 1.
Correctly retried sector 1677 in pass 1.
Correctly retried sector 1689 in pass 1.
Correctly retried sector 1700 in pass 1.
Correctly retried sector 1712 in pass 1.
Correctly retried sector 1743 in pass 1.
Correctly retried sector 1748 in pass 1.
Correctly retried sector 1755 in pass 1.
Correctly retried sector 1766 in pass 1.
Correctly retried sector 1778 in pass 1.
Correctly retried sector 1793 in pass 1.
Correctly retried sector 1794 in pass 1.
Correctly retried sector 1836 in pass 1.
Correctly retried sector 1845 in pass 1.
Correctly retried sector 1848 in pass 1.
Correctly retried sector 1859 in pass 1.
Correctly retried sector 1884 in pass 1.
Correctly retried sector 1902 in pass 1.
Correctly retried sector 1911 in pass 1.
Correctly retried sector 1925 in pass 1.
Correctly retried sector 1938 in pass 1.
Correctly retried sector 1977 in pass 1.
Correctly retried sector 1980 in pass 1.
Correctly retried sector 1991 in pass 1.
Correctly retried sector 2043 in pass 1.
Correctly retried sector 2046 in pass 1.
Correctly retried sector 2053 in pass 1.
Correctly retried sector 2056 in pass 1.
Correctly retried sector 2065 in pass 1.
Correctly retried sector 2083 in pass 1.
Correctly retried sector 2108 in pass 1.
Correctly retried sector 2119 in pass 1.
Correctly retried sector 2122 in pass 1.
Correctly retried sector 2131 in pass 1.
Correctly retried sector 2174 in pass 1.
Correctly retried sector 2188 in pass 1.
Correctly retried sector 2197 in pass 1.
Correctly retried sector 2208 in pass 1.
Correctly retried sector 2215 in pass 1.
Correctly retried sector 2254 in pass 1.
Correctly retried sector 2263 in pass 1.
Correctly retried sector 2273 in pass 1.
Correctly retried sector 2274 in pass 1.
Correctly retried sector 2333 in pass 1.
Correctly retried sector 2344 in pass 1.
Correctly retried sector 2351 in pass 1.
Correctly retried sector 2356 in pass 1.
Correctly retried sector 2399 in pass 1.
Correctly retried sector 2409 in pass 1.
Correctly retried sector 2410 in pass 1.
Correctly retried sector 2419 in pass 1.
Correctly retried sector 2422 in pass 1.
Correctly retried sector 2433 in pass 1.
Correctly retried sector 2436 in pass 1.
Correctly retried sector 2488 in pass 1.
Correctly retried sector 2499 in pass 1.
Correctly retried sector 2541 in pass 1.
Correctly retried sector 2542 in pass 1.
Correctly retried sector 2554 in pass 1.
Correctly retried sector 2564 in pass 1.
Correctly retried sector 2576 in pass 1.
Correctly retried sector 2623 in pass 1.
Correctly retried sector 2630 in pass 1.
Correctly retried sector 2669 in pass 1.
Correctly retried sector 2681 in pass 1.
Correctly retried sector 2682 in pass 1.
Correctly retried sector 2696 in pass 1.
Correctly retried sector 2703 in pass 1.
Correctly retried sector 2705 in pass 1.
Correctly retried sector 2708 in pass 1.
Correctly retried sector 2723 in pass 1.
Correctly retried sector 2762 in pass 1.
Correctly retried sector 2771 in pass 1.
Correctly retried sector 2774 in pass 1.
Correctly retried sector 2813 in pass 1.
Correctly retried sector 2841 in pass 1.
Correctly retried sector 2844 in pass 1.
Correctly retried sector 2859 in pass 1.
Correctly retried sector 2864 in pass 1.
Correctly retried sector 2907 in pass 1.
Correctly retried sector 2910 in pass 1.
Correctly retried sector 2917 in pass 1.
Correctly retried sector 2930 in pass 1.
Correctly retried sector 2944 in pass 1.
Correctly retried sector 2991 in pass 1.
Correctly retried sector 2996 in pass 1.
Correctly retried sector 3003 in pass 1.
Correctly retried sector 3010 in pass 1.
Correctly retried sector 3039 in pass 1.
Correctly retried sector 3049 in pass 1.
Correctly retried sector 3061 in pass 1.
Correctly retried sector 3062 in pass 1.
Correctly retried sector 3072 in pass 1.
Correctly retried sector 3079 in pass 1.
Correctly retried sector 3131 in pass 1.
Correctly retried sector 3137 in pass 1.
Correctly retried sector 3138 in pass 1.
Correctly retried sector 3180 in pass 1.
Correctly retried sector 3189 in pass 1.
Correctly retried sector 3211 in pass 1.
Correctly retried sector 3216 in pass 1.
Correctly retried sector 3234 in pass 1.
Correctly retried sector 3263 in pass 1.
Correctly retried sector 3269 in pass 1.
Correctly retried sector 3270 in pass 1.
Correctly retried sector 3282 in pass 1.
Correctly retried sector 3321 in pass 1.
Correctly retried sector 3324 in pass 1.
Correctly retried sector 3340 in pass 1.
Correctly retried sector 3352 in pass 1.
Correctly retried sector 3367 in pass 1.
Correctly retried sector 3370 in pass 1.
Correctly retried sector 3379 in pass 1.
Correctly retried sector 3405 in pass 1.
Correctly retried sector 3406 in pass 1.
Correctly retried sector 3418 in pass 1.
Correctly retried sector 3425 in pass 1.
Correctly retried sector 3484 in pass 1.
Correctly retried sector 3499 in pass 1.
Correctly retried sector 3502 in pass 1.
Correctly retried sector 3511 in pass 1.
Correctly retried sector 3550 in pass 1.
Correctly retried sector 3557 in pass 1.
Correctly retried sector 3560 in pass 1.
Correctly retried sector 3569 in pass 1.
Correctly retried sector 3587 in pass 1.
Correctly retried sector 3629 in pass 1.
Correctly retried sector 3630 in pass 1.
Correctly retried sector 3639 in pass 1.
Correctly retried sector 3642 in pass 1.
Correctly retried sector 3677 in pass 1.
Correctly retried sector 3688 in pass 1.
Correctly retried sector 3697 in pass 1.
Correctly retried sector 3700 in pass 1.
Correctly retried sector 3719 in pass 1.
Correctly retried sector 3731 in pass 1.
Correctly retried sector 3774 in pass 1.
Correctly retried sector 3777 in pass 1.
Correctly retried sector 3820 in pass 1.
Correctly retried sector 3832 in pass 1.
Correctly retried sector 3855 in pass 1.
Correctly retried sector 3860 in pass 1.
Correctly retried sector 3867 in pass 1.
Correctly retried sector 3878 in pass 1.
Correctly retried sector 3913 in pass 1.
Correctly retried sector 3926 in pass 1.
Correctly retried sector 3936 in pass 1.
Correctly retried sector 3965 in pass 1.
Correctly retried sector 3992 in pass 1.
Correctly retried sector 3999 in pass 1.
Correctly retried sector 4010 in pass 1.
Correctly retried sector 4019 in pass 1.
Correctly retried sector 4022 in pass 1.
Correctly retried sector 4045 in pass 1.
Correctly retried sector 4057 in pass 1.
Correctly retried sector 4058 in pass 1.
Correctly retried sector 4068 in pass 1.
Correctly retried sector 4127 in pass 1.
Correctly retried sector 4137 in pass 1.
Correctly retried sector 4150 in pass 1.
Correctly retried sector 4185 in pass 1.
Correctly retried sector 4196 in pass 1.
Correctly retried sector 4203 in pass 1.
Correctly retried sector 4208 in pass 1.
Correctly retried sector 4227 in pass 1.
Correctly retried sector 4230 in pass 1.
Correctly retried sector 4269 in pass 1.
Correctly retried sector 4282 in pass 1.
Correctly retried sector 4288 in pass 1.
Correctly retried sector 4317 in pass 1.
Correctly retried sector 4335 in pass 1.
Correctly retried sector 4340 in pass 1.
Correctly retried sector 4363 in pass 1.
Correctly retried sector 4366 in pass 1.
Correctly retried sector 4375 in pass 1.
Correctly retried sector 4386 in pass 1.
Correctly retried sector 4421 in pass 1.
Correctly retried sector 4433 in pass 1.
Correctly retried sector 4434 in pass 1.
Correctly retried sector 4476 in pass 1.
Correctly retried sector 4507 in pass 1.
Correctly retried sector 4517 in pass 1.
Correctly retried sector 4518 in pass 1.
Correctly retried sector 4530 in pass 1.
Correctly retried sector 4553 in pass 1.
Correctly retried sector 4556 in pass 1.
Correctly retried sector 4565 in pass 1.
Correctly retried sector 4576 in pass 1.
Correctly retried sector 4638 in pass 1.
Correctly retried sector 4645 in pass 1.
Correctly retried sector 4657 in pass 1.
Correctly retried sector 4658 in pass 1.
Correctly retried sector 4684 in pass 1.
Correctly retried sector 4693 in pass 1.
Correctly retried sector 4696 in pass 1.
Correctly retried sector 4711 in pass 1.
Correctly retried sector 4738 in pass 1.
Correctly retried sector 4777 in pass 1.
Correctly retried sector 4780 in pass 1.
Correctly retried sector 4789 in pass 1.
Correctly retried sector 4828 in pass 1.
Correctly retried sector 4843 in pass 1.
Correctly retried sector 4848 in pass 1.
Correctly retried sector 4855 in pass 1.
Correctly retried sector 4874 in pass 1.
Correctly retried sector 4883 in pass 1.
Correctly retried sector 4925 in pass 1.
Correctly retried sector 4926 in pass 1.
Correctly retried sector 4932 in pass 1.
Correctly retried sector 4984 in pass 1.
Correctly retried sector 4991 in pass 1.
Correctly retried sector 5006 in pass 1.
Correctly retried sector 5015 in pass 1.
Correctly retried sector 5018 in pass 1.
Correctly retried sector 5025 in pass 1.
Correctly retried sector 5064 in pass 1.
Correctly retried sector 5073 in pass 1.
Correctly retried sector 5091 in pass 1.
Correctly retried sector 5133 in pass 1.
Correctly retried sector 5146 in pass 1.
Correctly retried sector 5153 in pass 1.
Correctly retried sector 5156 in pass 1.
Correctly retried sector 5192 in pass 1.
Correctly retried sector 5199 in pass 1.
Correctly retried sector 5204 in pass 1.
Correctly retried sector 5219 in pass 1.
Correctly retried sector 5222 in pass 1.
Correctly retried sector 5277 in pass 1.
Correctly retried sector 5278 in pass 1.
Correctly retried sector 5288 in pass 1.
Correctly retried sector 5297 in pass 1.
Correctly retried sector 5336 in pass 1.
Correctly retried sector 5351 in pass 1.
Correctly retried sector 5354 in pass 1.
Correctly retried sector 5363 in pass 1.
Correctly retried sector 5382 in pass 1.
Correctly retried sector 5394 in pass 1.
Correctly retried sector 5433 in pass 1.
Correctly retried sector 5440 in pass 1.
Correctly retried sector 5487 in pass 1.
Correctly retried sector 5499 in pass 1.
Correctly retried sector 5513 in pass 1.
Correctly retried sector 5514 in pass 1.
Correctly retried sector 5526 in pass 1.
Correctly retried sector 5565 in pass 1.
Correctly retried sector 5572 in pass 1.
Correctly retried sector 5579 in pass 1.
Correctly retried sector 5584 in pass 1.
Correctly retried sector 5631 in pass 1.
Correctly retried sector 5641 in pass 1.
Correctly retried sector 5653 in pass 1.
Correctly retried sector 5654 in pass 1.
Correctly retried sector 5664 in pass 1.
Correctly retried sector 5707 in pass 1.
Correctly retried sector 5712 in pass 1.
Correctly retried sector 5719 in pass 1.
Correctly retried sector 5730 in pass 1.
Correctly retried sector 5773 in pass 1.
Correctly retried sector 5785 in pass 1.
Correctly retried sector 5796 in pass 1.
Correctly retried sector 5808 in pass 1.
Correctly retried sector 5839 in pass 1.
Correctly retried sector 5844 in pass 1.
Correctly retried sector 5851 in pass 1.
Correctly retried sector 5862 in pass 1.
Correctly retried sector 5874 in pass 1.
Correctly retried sector 5889 in pass 1.
Correctly retried sector 5890 in pass 1.
Correctly retried sector 5932 in pass 1.
Correctly retried sector 5941 in pass 1.
Correctly retried sector 5944 in pass 1.
Correctly retried sector 5955 in pass 1.
Correctly retried sector 5980 in pass 1.
Correctly retried sector 5998 in pass 1.
Correctly retried sector 6007 in pass 1.
Correctly retried sector 6021 in pass 1.
Correctly retried sector 6034 in pass 1.
Correctly retried sector 6073 in pass 1.
Correctly retried sector 6076 in pass 1.
Correctly retried sector 6087 in pass 1.
Correctly retried sector 6139 in pass 1.
Correctly retried sector 6142 in pass 1.
Correctly retried sector 6149 in pass 1.
Correctly retried sector 6152 in pass 1.
Correctly retried sector 6161 in pass 1.
Correctly retried sector 6179 in pass 1.
Correctly retried sector 6204 in pass 1.
Correctly retried sector 6215 in pass 1.
Correctly retried sector 6218 in pass 1.
Correctly retried sector 6227 in pass 1.
Correctly retried sector 6270 in pass 1.
Correctly retried sector 6284 in pass 1.
Correctly retried sector 6293 in pass 1.
Correctly retried sector 6304 in pass 1.
Correctly retried sector 6311 in pass 1.
Correctly retried sector 6350 in pass 1.
Correctly retried sector 6359 in pass 1.
Correctly retried sector 6369 in pass 1.
Correctly retried sector 6370 in pass 1.
Correctly retried sector 6429 in pass 1.
Correctly retried sector 6440 in pass 1.
Correctly retried sector 6447 in pass 1.
Correctly retried sector 6452 in pass 1.
Correctly retried sector 6495 in pass 1.
Correctly retried sector 6505 in pass 1.
Correctly retried sector 6506 in pass 1.
Correctly retried sector 6515 in pass 1.
Correctly retried sector 6518 in pass 1.
Correctly retried sector 6529 in pass 1.
Correctly retried sector 6532 in pass 1.
Correctly retried sector 6584 in pass 1.
Correctly retried sector 6595 in pass 1.
Correctly retried sector 6637 in pass 1.
Correctly retried sector 6638 in pass 1.
Correctly retried sector 6650 in pass 1.
Correctly retried sector 6660 in pass 1.
Correctly retried sector 6672 in pass 1.
Correctly retried sector 6719 in pass 1.
Correctly retried sector 6726 in pass 1.
Correctly retried sector 6765 in pass 1.
Correctly retried sector 6777 in pass 1.
Correctly retried sector 6778 in pass 1.
Correctly retried sector 6792 in pass 1.
Correctly retried sector 6799 in pass 1.
Correctly retried sector 6801 in pass 1.
Correctly retried sector 6804 in pass 1.
Correctly retried sector 6819 in pass 1.
Correctly retried sector 6858 in pass 1.
Correctly retried sector 6867 in pass 1.
Correctly retried sector 6870 in pass 1.
Correctly retried sector 6909 in pass 1.
Correctly retried sector 6937 in pass 1.
Correctly retried sector 6940 in pass 1.
Correctly retried sector 6955 in pass 1.
Correctly retried sector 6960 in pass 1.
Correctly retried sector 7003 in pass 1.
Correctly retried sector 7006 in pass 1.
Correctly retried sector 7013 in pass 1.
Correctly retried sector 7026 in pass 1.
Correctly retried sector 7040 in pass 1.
Correctly retried sector 7087 in pass 1.
Correctly retried sector 7092 in pass 1.
Correctly retried sector 7099 in pass 1.
Correctly retried sector 7106 in pass 1.
Correctly retried sector 7135 in pass 1.
Correctly retried sector 7145 in pass 1.
Correctly retried sector 7157 in pass 1.
Correctly retried sector 7158 in pass 1.
Correctly retried sector 7168 in pass 1.
Correctly retried sector 7175 in pass 1.
Correctly retried sector 7227 in pass 1.
Correctly retried sector 7233 in pass 1.
Correctly retried sector 7234 in pass 1.
Correctly retried sector 7276 in pass 1.
Correctly retried sector 7285 in pass 1.
Correctly retried sector 7307 in pass 1.
Correctly retried sector 7312 in pass 1.
Correctly retried sector 7330 in pass 1.
Correctly retried sector 7359 in pass 1.
Correctly retried sector 7365 in pass 1.
Correctly retried sector 7366 in pass 1.
Correctly retried sector 7378 in pass 1.
Correctly retried sector 7417 in pass 1.
Correctly retried sector 7420 in pass 1.
Correctly retried sector 7436 in pass 1.
Correctly retried sector 7448 in pass 1.
Correctly retried sector 7463 in pass 1.
Correctly retried sector 7466 in pass 1.
Correctly retried sector 7475 in pass 1.
Correctly retried sector 7501 in pass 1.
Correctly retried sector 7502 in pass 1.
Correctly retried sector 7514 in pass 1.
Correctly retried sector 7521 in pass 1.
Correctly retried sector 7580 in pass 1.
Correctly retried sector 7595 in pass 1.
Correctly retried sector 7598 in pass 1.
Correctly retried sector 7607 in pass 1.
Correctly retried sector 7646 in pass 1.
Correctly retried sector 7653 in pass 1.
Correctly retried sector 7656 in pass 1.
Correctly retried sector 7665 in pass 1.
Correctly retried sector 7683 in pass 1.
Correctly retried sector 7725 in pass 1.
Correctly retried sector 7726 in pass 1.
Correctly retried sector 7735 in pass 1.
Correctly retried sector 7738 in pass 1.
Correctly retried sector 7773 in pass 1.
Correctly retried sector 7784 in pass 1.
Correctly retried sector 7793 in pass 1.
Correctly retried sector 7796 in pass 1.
Correctly retried sector 7815 in pass 1.
Correctly retried sector 7827 in pass 1.
Correctly retried sector 7870 in pass 1.
Correctly retried sector 7873 in pass 1.
Correctly retried sector 7916 in pass 1.
Correctly retried sector 7928 in pass 1.
Correctly retried sector 7951 in pass 1.
Correctly retried sector 7956 in pass 1.
Correctly retried sector 7963 in pass 1.
Correctly retried sector 7974 in pass 1.
Correctly retried sector 8009 in pass 1.
Correctly retried sector 8022 in pass 1.
Correctly retried sector 8032 in pass 1.
Correctly retried sector 8061 in pass 1.
Correctly retried sector 8088 in pass 1.
Correctly retried sector 8095 in pass 1.
Correctly retried sector 8106 in pass 1.
Correctly retried sector 8115 in pass 1.
Correctly retried sector 8118 in pass 1.
Correctly retried sector 8141 in pass 1.
Correctly retried sector 8153 in pass 1.
Correctly retried sector 8154 in pass 1.
Correctly retried sector 8164 in pass 1.
Correctly retried sector 8223 in pass 1.
Correctly retried sector 8233 in pass 1.
Correctly retried sector 8246 in pass 1.
Correctly retried sector 8281 in pass 1.
Correctly retried sector 8292 in pass 1.
Correctly retried sector 8299 in pass 1.
Correctly retried sector 8304 in pass 1.
Correctly retried sector 8323 in pass 1.
Correctly retried sector 8326 in pass 1.
Correctly retried sector 8365 in pass 1.
Correctly retried sector 8378 in pass 1.
Correctly retried sector 8384 in pass 1.
Correctly retried sector 8413 in pass 1.
Correctly retried sector 8431 in pass 1.
Correctly retried sector 8436 in pass 1.
Correctly retried sector 8459 in pass 1.
Correctly retried sector 8462 in pass 1.
Correctly retried sector 8471 in pass 1.
Correctly retried sector 8482 in pass 1.
Correctly retried sector 8517 in pass 1.
Correctly retried sector 8529 in pass 1.
Correctly retried sector 8530 in pass 1.
Correctly retried sector 8572 in pass 1.
Correctly retried sector 8603 in pass 1.
Correctly retried sector 8613 in pass 1.
Correctly retried sector 8614 in pass 1.
Correctly retried sector 8626 in pass 1.
Correctly retried sector 8649 in pass 1.
Correctly retried sector 8652 in pass 1.
Correctly retried sector 8661 in pass 1.
Correctly retried sector 8672 in pass 1.
Correctly retried sector 8734 in pass 1.
Correctly retried sector 8741 in pass 1.
Correctly retried sector 8753 in pass 1.
Correctly retried sector 8754 in pass 1.
Correctly retried sector 8780 in pass 1.
Correctly retried sector 8789 in pass 1.
Correctly retried sector 8792 in pass 1.
Correctly retried sector 8807 in pass 1.
Correctly retried sector 8834 in pass 1.
Correctly retried sector 8873 in pass 1.
Correctly retried sector 8876 in pass 1.
Correctly retried sector 8885 in pass 1.
Correctly retried sector 8924 in pass 1.
Correctly retried sector 8939 in pass 1.
Correctly retried sector 8944 in pass 1.
Correctly retried sector 8951 in pass 1.
Correctly retried sector 8970 in pass 1.
Correctly retried sector 8979 in pass 1.
Correctly retried sector 9021 in pass 1.
Correctly retried sector 9022 in pass 1.
Correctly retried sector 9028 in pass 1.
Correctly retried sector 9080 in pass 1.
Correctly retried sector 9087 in pass 1.
Correctly retried sector 9102 in pass 1.
Correctly retried sector 9111 in pass 1.
Correctly retried sector 9114 in pass 1.
Correctly retried sector 9121 in pass 1.
Correctly retried sector 9160 in pass 1.
Correctly retried sector 9169 in pass 1.
Correctly retried sector 9187 in pass 1.
Correctly retried sector 9229 in pass 1.
Correctly retried sector 9242 in pass 1.
Correctly retried sector 9249 in pass 1.
Correctly retried sector 9252 in pass 1.
Correctly retried sector 9288 in pass 1.
Correctly retried sector 9295 in pass 1.
Correctly retried sector 9300 in pass 1.
Correctly retried sector 9315 in pass 1.
Correctly retried sector 9318 in pass 1.
Correctly retried sector 9373 in pass 1.
Correctly retried sector 9374 in pass 1.
Correctly retried sector 9384 in pass 1.
Correctly retried sector 9393 in pass 1.
Correctly retried sector 9432 in pass 1.
Correctly retried sector 9447 in pass 1.
Correctly retried sector 9450 in pass 1.
Correctly retried sector 9459 in pass 1.
Correctly retried sector 9478 in pass 1.
Correctly retried sector 9490 in pass 1.
Correctly retried sector 9529 in pass 1.
Correctly retried sector 9536 in pass 1.
Correctly retried sector 9583 in pass 1.
Correctly retried sector 9595 in pass 1.
Correctly retried sector 9609 in pass 1.
Correctly retried sector 9610 in pass 1.
Correctly retried sector 9622 in pass 1.
Correctly retried sector 9661 in pass 1.
Correctly retried sector 9668 in pass 1.
Correctly retried sector 9675 in pass 1.
Correctly retried sector 9680 in pass 1.
Correctly retried sector 9727 in pass 1.
Correctly retried sector 9737 in pass 1.
Correctly retried sector 9749 in pass 1.
Correctly retried sector 9750 in pass 1.
Correctly retried sector 9760 in pass 1.
Correctly retried sector 9803 in pass 1.
Correctly retried sector 9808 in pass 1.
Correctly retried sector 9815 in pass 1.
Correctly retried sector 9826 in pass 1.
Correctly retried sector 9869 in pass 1.
Correctly retried sector 9881 in pass 1.
Correctly retried sector 9892 in pass 1.
Correctly retried sector 9904 in pass 1.
Correctly retried sector 9935 in pass 1.
Correctly retried sector 9940 in pass 1.
Correctly retried sector 9947 in pass 1.
Correctly retried sector 9958 in pass 1.
Correctly retried sector 9970 in pass 1.
Correctly retried sector 9985 in pass 1.
Correctly retried sector 9986 in pass 1.
Correctly retried sector 10028 in pass 1.
Correctly retried sector 10037 in pass 1.
Correctly retried sector 10040 in pass 1.
Correctly retried sector 10051 in pass 1.
Correctly retried sector 10076 in pass 1.
Correctly retried sector 10094 in pass 1.
Correctly retried sector 10103 in pass 1.
Correctly retried sector 10130 in pass 1.
Correctly retried sector 10169 in pass 1.
Correctly retried sector 10172 in pass 1.
Correctly retried sector 10183 in pass 1.


Correctly retried sector 1733 subchannel in pass 1.
Retrying sector 19460 subchannel, pass 3, forward
Closing output file.
Closed in 1.8038097 seconds.
Hashing image file byte 32505856 of 32853662
Hashing media tags...
Checking filesystems
Hashing tracks...
Hashing sector 19456 of 19527
Hashing subchannel sector 19456 of 19527

Writing metadata sidecar

Took a total of 2166.611 seconds (2794.845 processing commands, 17.407 checksumming, 7.975 writing, 1.804 closing).
Average speed: 0.016 MiB/sec.
Fastest speed burst: 21.288 MiB/sec.
Slowest speed burst: 0.016 MiB/sec.
0 sectors could not be read.
10 subchannels could not be read.
```

### Media details

Confirmed to occur with http://redump.org/disc/91538/ and http://redump.org/disc/25579/

**Comment by TheRogueArchivist** [comment]
Issue still present on newest current commit of devel branch [82f2fd6](https://github.com/aaru-dps/Aaru/commit/82f2fd6396caae71a19371c8955b88ea07381f13).

**Comment by lotharsm** [comment]
I can reproduce this on an ASUS BW-16D1HT as well, using both 6.0.0alpha9 and a self-compiled binary as of https://github.com/aaru-dps/Aaru/commit/4825d414604fe30e23ceabc169be94f1b0dcd781

I'm able to trigger this bug on a disc with intentionally bad sectors (ring protection), during the re-reads, it fixes the sector even if they report media errors in the first run.

**Comment by lotharsm** [comment]
I agree with @TheRogueArchivist that this is likely a cache issue.

I can somewhat replicate this issue with this drive (tested 3.02, 3.10 and 3.10 with RipShark mod) on redumper as well:

```
[LBA:   1080] C2 error (bits:  344, data crc: E858835F, C2 crc: 3D179DF2, retry: 1)
[LBA:   1080] correction failure
[LBA:   1081] C2 error (bits: 1992, data crc: 8140E0AF, C2 crc: 1109C4DF, retry: 1)
[LBA:   1081] correction failure
[LBA:   1082] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1082] correction failure
[LBA:   1083] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1083] correction failure
[LBA:   1084] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1084] correction failure
[LBA:   1085] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1085] correction failure
[LBA:   1086] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1086] correction failure
[LBA:   1087] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1087] correction failure
[LBA:   1088] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1088] correction failure
[LBA:   1089] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1089] correction failure
[LBA:   1090] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1090] correction failure
[LBA:   1091] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1091] correction failure
[LBA:   1092] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1092] correction failure
[LBA:   1093] C2 error (bits: 2352, data crc: D138F4D3, C2 crc: 4F1A7F6B, retry: 1)
[LBA:   1093] correction failure
[LBA:   1094] SCSI error (SC: CHECK CONDITION, SK: MEDIUM ERROR, ASC: UNRECOVERED READ ERROR, ASCQ: 05, retry: 1)
[LBA:   1094] correction failure
[LBA:   1095] SCSI error (SC: CHECK CONDITION, SK: HARDWARE ERROR, ASC: LOGICAL UNIT HAS NOT SELF-CONFIGURED YET, ASCQ: 02, retry: 1)
[LBA:   1095] correction failure
[LBA:   1096] SCSI error (SC: CHECK CONDITION, SK: MEDIUM ERROR, ASC: NO SEEK COMPLETE, retry: 1)
[LBA:   1096] correction failure
[LBA:   1097] SCSI error (SC: CHECK CONDITION, SK: MEDIUM ERROR, ASC: NO SEEK COMPLETE, retry: 1)
[LBA:   1097] correction failure
[LBA:   1098] SCSI error (SC: CHECK CONDITION, SK: MEDIUM ERROR, ASC: NO SEEK COMPLETE, retry: 1)
[LBA:   1098] correction failure
[LBA:   1099] SCSI error (SC: CHECK CONDITION, SK: MEDIUM ERROR, ASC: NO SEEK COMPLETE, retry: 1)
```

The problem is that starting from LBA 1080, the disc contains a ring protection that doesn't even carry a sync signal, so the drive shouldn't be able to read any data from it at all.

I am not sure if this is the exact same issue though since redumper uses BE_CDDA on this drive.

**Comment by claunia** [comment]
Fixed in 97827d9bfe0750dbcb028416ef4cd80b26c4fb60
